src/splice.ts

```diff
@@ -1,12 +1,133 @@
+import type {
+  ArraySplice,
+  GreaterThan,
+  IfNever,
+  IsLiteral,
+  IsNegative,
+  Subtract,
+  Sum,
+} from "type-fest";
+import type {
+  CoercedArray,
+  IterableContainer,
+  TupleParts,
+  WidenLiteral,
+} from "./internal/types";
 import { purry } from "./purry";
 
+type IfLiteral<N extends number, Fallback> = number extends N ? Fallback : N;
+
+type PositiveIndex<
+  T extends IterableContainer,
+  N extends number,
+> = N extends unknown
+  ? IsNegative<N> extends false
+    ? GreaterThan<N, T["length"]> extends true
+      ? T["length"]
+      : N
+    : number extends T["length"]
+      ? number
+      : IfLiteral<Sum<T["length"], N>, 0>
```

Comment:
Can you document why 0 is the fallback? It's not clear to me; intuitively, it feels like it should be `never`, but I don't know without an explanation.